cpp/src/arrow/ArrowConfig.cmake.in

```diff
@@ -172,3 +172,16 @@ endmacro()
 arrow_keep_backward_compatibility(Arrow arrow)
 
 check_required_components(Arrow)
+
+function(arrow_show_details PACKAGE_NAME)
+  if(NOT ${PACKAGE_NAME}_FIND_QUIETLY)
+    string(TOUPPER ${PACKAGE_NAME} package_name_upper)
+    message(STATUS "${PACKAGE_NAME} version: ${${package_name_upper}_VERSION}")
+    message(STATUS "Found the ${PACKAGE_NAME} shared library: ${${package_name_upper}_SHARED_LIB}")
```

Comment:
Does this work with `ArrowDataset`?

It seems that we need to pass 2 arguments to this function: `arrow_show_details(package_name variable_prefix)` (e.g.: `arrow_show_details(ArrowDataset ARROW_DATASET)`)

Reply:
The signature now is `macro(arrow_show_details package_name variable_prefix)`, but does it make sense to use `target_base_name` instead of `variable_prefix` as `arrow_keep_backward_compatibility`, and do something like:
```
macro(arrow_show_details package_name target_base_name)
  string(TOUPPER ${target_base_name} target_base_name_upper)
  .
  .
  .
endmacro()
```